src/daft-micropartition/src/micropartition.rs

```diff
@@ -588,7 +594,12 @@ impl MicroPartition {
                     .map(|s| s.get_parquet_metadata().cloned())
                     .collect::<Option<Vec<_>>>();
 
-                let row_groups = parquet_sources_to_row_groups(scan_task.sources.as_slice());
+                let row_groups = if let Some(rgs) = row_groups {
+                    Some(rgs.clone())
+                } else {
+                    parquet_sources_to_row_groups(scan_task.sources.as_slice())
+                };
```

Comment:
Found the issue!

So at this point, `row_groups` is the rowgroups specified by the user at a top-level for this Parquet file.

However, the ScanTask we are trying to materialize might have a different set of rowgroups (after we perform rowgroup splitting).

Hence we should prefer `parquet_sources_to_row_groups(scan_task.sources.as_slice())` instead of `row_groups` when available.

Reply:
To get the correct behavior, I think we need to:

1. Leave this line unchanged: `let row_groups = parquet_sources_to_row_groups(scan_task.sources.as_slice());`. I.e. when materializing the ScanTask, use the ScanTask's current chunk spec as the source of truth as to what we need to materialize.
2. Instead, we should change our logic for creating the ScanTasks in `GlobScanOperator` to create each ScanTask with the appropriate rowgroups there, so we get the appropriate splitting behavior

Reply:
This is where we need to add information onto each ScanTask about what rowgroups we want to read (in `GlobScanOperator::to_scan_task`):

<img width="580" alt="image" src="https://github.com/user-attachments/assets/ac8ae857-e10a-4f75-b9b7-99ffc8fc1ad2">